EL7 : presence of libmacaroons-devel breaks xrootd compilation because of alidist uuid

When libmacaroons-devel is detected by xrootd and it is enabled, the compilation breaks with:
```
DEBUG:XRootD:XRootD:v4.11.1: [ 98%] Linking CXX shared library libXrdFfs.so
DEBUG:XRootD:XRootD:v4.11.1: /home/physics-tools/alicebuild/sw/SOURCES/XRootD/v4.11.1/v4.11.1/src/XrdMacaroons/XrdMacaroonsHandler.cc: In member function 'std::string Macaroons::Handler::GenerateID(const string&, const XrdSecEntity&, const string&, const std::vector<std::__cxx11::basic_string<char> >&, const string&)':
DEBUG:XRootD:XRootD:v4.11.1: /home/physics-tools/alicebuild/sw/SOURCES/XRootD/v4.11.1/v4.11.1/src/XrdMacaroons/XrdMacaroonsHandler.cc:114:12: error: aggregate 'uuid_t uu' has incomplete type and cannot be defined
DEBUG:XRootD:XRootD:v4.11.1:   114 |     uuid_t uu;
DEBUG:XRootD:XRootD:v4.11.1:       |            ^~
DEBUG:XRootD:XRootD:v4.11.1: /home/physics-tools/alicebuild/sw/SOURCES/XRootD/v4.11.1/v4.11.1/src/XrdMacaroons/XrdMacaroonsHandler.cc:115:5: error: 'uuid_generate_random' was not declared in this scope
DEBUG:XRootD:XRootD:v4.11.1:   115 |     uuid_generate_random(uu);
DEBUG:XRootD:XRootD:v4.11.1:       |     ^~~~~~~~~~~~~~~~~~~~
DEBUG:XRootD:XRootD:v4.11.1: /home/physics-tools/alicebuild/sw/SOURCES/XRootD/v4.11.1/v4.11.1/src/XrdMacaroons/XrdMacaroonsHandler.cc:117:5: error: 'uuid_unparse' was not declared in this scope; did you mean 'uuid_compare'?
DEBUG:XRootD:XRootD:v4.11.1:   117 |     uuid_unparse(uu, uuid_buf);
DEBUG:XRootD:XRootD:v4.11.1:       |     ^~~~~~~~~~~~
DEBUG:XRootD:XRootD:v4.11.1:       |     uuid_compare
DEBUG:XRootD:XRootD:v4.11.1: gmake[2]: *** [src/CMakeFiles/XrdMacaroons-4.dir/build.make:76: src/CMakeFiles/XrdMacaroons-4.dir/XrdMacaroons/XrdMacaroonsHandler.cc.o] Error 1
```
log files can be found [here](http://issaf.spacescience.ro/adrian/alibuild/xrootd/)

--- a/xrootd.sh
+++ b/xrootd.sh
@@ -30,7 +30,9 @@ case $ARCHITECTURE in
 esac
 
 rsync -a --exclude '**/.git' --delete $SOURCEDIR/ $BUILDDIR
-sed -i.bak 's/"uuid.h"/"uuid\/uuid.h"/' $(find . -name "*Macaroon*Handler*.cc")
+
+[[ x"$XROOTD_V4" == x"True" ]] && sed -i.bak 's/"uuid.h"/"uuid\/uuid.h"/' $(find . -name "*Macaroon*Handler*.cc")
+
 
 mkdir build
 pushd build

--- a/xrootd.sh
+++ b/xrootd.sh
@@ -29,7 +29,7 @@ case $ARCHITECTURE in
   ;;
 esac
 
-rsync -a --exclude '**/.git' --delete $SOURCEDIR/ $BUILDDIR
+rsync -a --delete $SOURCEDIR/ $BUILDDIR
 
 [ x"$XROOTD_V4" = x"True" ] && sed -i.bak 's/"uuid.h"/"uuid\/uuid.h"/' $(find . -name "*Macaroon*Handler*.cc")
 

--- a/xrootd.sh
+++ b/xrootd.sh
@@ -29,7 +29,12 @@ case $ARCHITECTURE in
   ;;
 esac
 
-cmake "$SOURCEDIR"                                                    \
+rsync -a --exclude '**/.git' --delete $SOURCEDIR/ $BUILDDIR
+sed -i.bak 's/"uuid.h"/"uuid\/uuid.h"/' $(find . -name "*Macaroon*Handler*.cc")
+
+mkdir build
+pushd build
+cmake "$BUILDDIR"                                                     \
       ${CMAKE_GENERATOR:+-G "$CMAKE_GENERATOR"}                       \
       -DCMAKE_INSTALL_PREFIX=$INSTALLROOT                             \
       -DCMAKE_INSTALL_LIBDIR=lib                                      \
@@ -47,6 +52,7 @@ cmake "$SOURCEDIR"                                                    \
       -DCMAKE_CXX_FLAGS_RELWITHDEBINFO="-Wno-error"
 
 cmake --build . -- ${JOBS:+-j$JOBS} install
+popd
 
 if [[ x"$XROOTD_PYTHON" == x"True" ]];
 then

--- a/xrootd.sh
+++ b/xrootd.sh
@@ -31,7 +31,7 @@ esac
 
 rsync -a --exclude '**/.git' --delete $SOURCEDIR/ $BUILDDIR
 
-[[ x"$XROOTD_V4" == x"True" ]] && sed -i.bak 's/"uuid.h"/"uuid\/uuid.h"/' $(find . -name "*Macaroon*Handler*.cc")
+[ x"$XROOTD_V4" = x"True" ] && sed -i.bak 's/"uuid.h"/"uuid\/uuid.h"/' $(find . -name "*Macaroon*Handler*.cc")
 
 
 mkdir build

